src/rules/graphql/no-more-than-100-fields.ts

```diff
@@ -0,0 +1,408 @@
+                                                }
+                                                Field90 {
+                                                    value
+                                                }
+                                                Field91 {
+                                                    value
+                                                }
+                                                Field92 {
+                                                    value
+                                                }
+                                                Field93 {
+                                                    value
+                                                }
+                                                Field94 {
+                                                    value
+                                                }
+                                                Field95 {
+                                                    value
+                                                }
+                                                Field96 {
+                                                    value
+                                                }
+                                                Field97 {
+                                                    value
+                                                }
+                                                Field98 {
+                                                    value
+                                                }
+                                                Field99 {
+                                                    value
+                                                }
+                                                Field100 {
+                                                    value
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    `
+                }
+            ]
+        },
+        messages: {
+            [NO_MORE_THAN_100_FIELDS_RULE_ID]:
+                'Offline GraphQL: The "{{entityName}}" entity contains {{numberOfFields}} fields. If an entity has more than 100 fields, the server might not return all the expected records.'
+        },
+        schema: []
+    },
+
+    create(context: GraphQLESLintRuleContext) {
+        return {
+            Field(node) {
+                // report lint issue if the graphql is like '... uiapi { aggregate { ...'
+                if (node.name.value === 'node') {
+                    const childSelectionSet = node.selectionSet;
+                    if (
+                        childSelectionSet !== undefined &&
+                        childSelectionSet.selections.length > 100
```

Comment:
My understanding was that we were looking for 100 fields _in the aggregate_ of the query. Can we double-check that assumption? I thought from our discussion, it came down to the total number of fields across all entities in the query.